File: trickysum.cpp
#include <cmath>
#include <stdio.h>
#include <iostream>
#include <algorithm>
#include <math.h>
#include <string>
using namespace std;

int main()
{
	int t;
	cin>>t;
	while(t--)
	{
		long long int a;
		cin>>a;
		long long int cost = 0;
		int total = (a*(a+1))/2;
		long long int itr=0;
		long long int x=0;
		while(itr<a)
		{
			itr = pow(2,x);
			total = total - 2*itr;
			x++;
		}
		cout<<total<<endl;
	}
	return 0;
}

File: trickysum.py
t = int(raw_input())
while (t>0):
	n = int(raw_input())
	total = n*(n+1)/2
	itr = 0
	x = 0
	while (True):
		itr = 2**(x)
		if(itr>n):
			break
		total = total- 2*itr
		x = x+1
	print total
	t = t-1